ui/app/components/token-editor.js

```diff
@@ -96,7 +134,15 @@ export default class TokenEditorComponent extends Component {
         this.activeToken.expirationTTL = null;
       }
 
-      await this.activeToken.save();
+      const adapterRegion = this.activeToken.global
+        ? this.system.get('defaultRegion.region')
+        : this.tokenRegion;
```

Comment:
Note: even though the token is marked `Global: true`, I found that I still have to unset its region (or set it to the authoritative region), otherwise it gets created as global but within whatever region the user happens to be using at the time.

Once you're browsing the web UI with a region set, all requests get a queryParam of ?region=<region>. Thus you could end up with a token that, yes, is global, but is stored in the context of an agent that isn't global/authoritative.

As such it seems like you _can_ create a global, non-global token. Weird!

Reply:
> Thus you could end up with a token that, yes, is global, but is stored in the context of an agent that isn't global/authoritative.

Depending on what you mean by the "context" of an agent, that smells like a possible bug in the RPC handler. We have logic (ref [`acl_endpoint.go#L602-L611`](https://github.com/hashicorp/nomad/blob/v1.8.1/nomad/acl_endpoint.go#L602-L611)) to override the region flag if the token is global.